salon_creator/lib/widgets/custom_dialog.dart

```diff
@@ -0,0 +1,73 @@
+import 'package:flutter/material.dart';
+
+void showCustomDialog({
+  title,
+  content,
+  leftFunction,
+  leftButtonText,
+  rightFunction,
+  rightButtonText,
+  context,
+}) {
+  showDialog(
+    context: context,
+    builder: (_) {
+      return AlertDialog(
+        contentPadding: EdgeInsets.zero,
+        title: title,
+        content: Column(
+          mainAxisSize: MainAxisSize.min,
+          children: [
+            _buildDialogText(content),
+            Container(
+              color: Theme.of(context).primaryColor,
+              child: _buildDialogButtons(
+                leftFunction,
+                leftButtonText,
+                rightFunction,
+                rightButtonText,
+              ),
+            ),
+          ],
+        ),
+      );
+    },
+  );
+}
+
+Container _buildDialogText(content) {
```

Comment:
Return types of all methods that return widgets (all across the code base) needs to be Widget, since this is the universal base class. It would be easier to change the return values later on without needing to change the method definition each time. Please apply this change to _all_ such functions in this PR.